internal/mode/static/nginx/agent/grpc/interceptor/interceptor.go

```diff
@@ -0,0 +1,83 @@
+package interceptor
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	grpcContext "github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/agent/grpc/context"
+)
+
+type ContextSetter struct{}
+
+func NewContextSetter() *ContextSetter {
```

Comment:
Do we need to return a pointer here?

Reply:
It can implement an interface without being a pointer...

Reply:
Oh, duh. I had all the methods with pointer receivers, so the Interceptor initialization needed a pointer type.